pontoon/base/views.py

```diff
@@ -294,19 +300,36 @@ def get_translations_from_other_locales(request):
         }, status=400)
 
     entity = get_object_or_404(Entity, pk=entity)
-    locales = entity.resource.project.locales.exclude(code=locale)
+    locale = get_object_or_404(Locale, code=locale)
+    selected_locales = request.user.profile.sorted_locales
     plural_form = None if entity.string_plural == "" else 0
 
+    preferred_source_translations = Translation.objects.filter(
+        entity=entity,
+        locale__in=selected_locales,
+        plural_form=plural_form,
+        approved=True
+    ).exclude(locale=locale)
+
+    preferred_locales = list(get_translation_values(preferred_source_translations))
+
     translations = Translation.objects.filter(
         entity=entity,
-        locale__in=locales,
         plural_form=plural_form,
         approved=True
-    ).order_by('locale__name')
+    ).exclude(locale=locale)
+
+    other_locales = list(get_translation_values(translations))
+
+    preferred_locale_codes = [element['locale__code'] for element in preferred_locales]
+    for code in preferred_locale_codes:
+        other_locales = [l for l in other_locales if not (l['locale__code'] == code)]
```

Comment:
You can drop these three lines if you define `preferred/other` as described above.